include/z64collision_check.h

```diff
@@ -8,114 +8,168 @@
 
 struct Actor;
 
+/*
+ * Bases for all shapes of colliders
+ */
+
+typedef enum {
+    /*  0 */ COLTYPE_HIT0, // Blue blood, white hitmark
+    /*  1 */ COLTYPE_HIT1, // No blood, dust hitmark
+    /*  2 */ COLTYPE_HIT2, // Green blood, dust hitmark
+    /*  3 */ COLTYPE_HIT3, // No blood, white hitmark
+    /*  4 */ COLTYPE_HIT4, // Water burst, no hitmark
+    /*  5 */ COLTYPE_HIT5, // No blood, red hitmark
+    /*  6 */ COLTYPE_HIT6, // Green blood, white hitmark
+    /*  7 */ COLTYPE_HIT7, // Red blood, white hitmark
+    /*  8 */ COLTYPE_HIT8, // Blue blood, red hitmark
+    /*  9 */ COLTYPE_METAL,
+    /* 10 */ COLTYPE_NONE,
+    /* 11 */ COLTYPE_WOOD,
+    /* 12 */ COLTYPE_HARD,
+    /* 13 */ COLTYPE_TREE
+} ColliderType;
+
+typedef enum {
+    /* 0 */ COLSHAPE_JNTSPH,
+    /* 1 */ COLSHAPE_CYLINDER,
+    /* 2 */ COLSHAPE_TRIS,
+    /* 3 */ COLSHAPE_QUAD,
+    /* 4 */ COLSHAPE_MAX
+} ColliderShape;
+
 typedef struct {
     /* 0x00 */ struct Actor* actor; // Attached actor
     /* 0x04 */ struct Actor* at; // Actor attached to what it collided with as an AT collider.
     /* 0x08 */ struct Actor* ac; // Actor attached to what it collided with as an AC collider.
     /* 0x0C */ struct Actor* oc; // Actor attached to what it collided with as an OC collider.
-    /* 0x10 */ u8 atFlags; // Information flags for AT collisions.
-    /* 0x11 */ u8 acFlags; // Information flags for AC collisions.
-    /* 0x12 */ u8 ocFlags1; // Information flags for OC collisions.
-    /* 0x13 */ u8 ocFlags2;  // Flags related to which colliders it can OC collide with.
-    /* 0x14 */ u8 colType; // Determines hitmarks and sound effects during AC collisions.
-    /* 0x15 */ u8 shape; // JntSph, Cylinder, Tris, or Quad
+    /* 0x10 */ u8 atFlags;
+    /* 0x11 */ u8 acFlags;
+    /* 0x12 */ u8 ocFlags1;
+    /* 0x13 */ u8 ocFlags2; // Flags related to which colliders it can OC collide with.
+    /* 0x14 */ u8 colType; // Determines hitmarks and sound effects during AC collisions. See `ColliderType` enum
```

Comment:
Probably not in this PR, but thoughts on shortening `colType` to just `type`. Seems weird type gets the qualifier, but `shape` doesn't

Reply:
strong disagree. If any property in Collider were to be called type, `shape` would be it, since the shape of a collider is it's most distinguishing feature; calling `colType` simply `type` would arguably be confusing and misleading. 

The way I see it, `colType`/ColliderType is just a placeholder name until we have a more complete understanding of the ColliderType enum's purpose, and so the name is fine as is.

Reply:
yeah this field isn't as important as "type" may make it sound

an appropriate name may or may not be `material` or `texture`